plugins/blockchain_statistics/statistics_sender.cpp

```diff
@@ -15,22 +15,32 @@
 #include <algorithm>
 
 
-statistics_sender::statistics_sender(uint32_t default_port) : default_port(default_port) {
+statistics_sender::statistics_sender() :
+    ios( appbase::app().get_io_service() ),
+    is_previous_bucket_set(false) {
+}
+
+statistics_sender::statistics_sender(uint32_t default_port) :
+    default_port(default_port),
+    ios( appbase::app().get_io_service() ),
+    is_previous_bucket_set(false) {
 }
 
 bool statistics_sender::can_start() {
     return !recipient_endpoint_set.empty();
 }
 
 void statistics_sender::push(const std::string & str) {
-    boost::asio::io_service io_service;
+    ios.post ([this, str]() {
+        boost::asio::ip::udp::socket socket(this->ios, boost::asio::ip::udp::endpoint(boost::asio::ip::udp::v4(), 0));
```

Comment:
You shouldn't create this socket on each push.
Move this socket to statistics_sender.